src/uct/ib/rc/accel/rc_mlx5_common.h

```diff
@@ -606,7 +606,6 @@ uct_rc_mlx5_txqp_dptr_post(uct_rc_iface_t *iface, enum ibv_qp_type qp_type,
     case MLX5_OPCODE_SEND:
         /* Data segment only */
         ucs_assert(length < (2ul << 30));
-        ucs_assert(length <= iface->super.config.seg_size);
```

Comment:
this assert is no more relevant because DM length may be over seg_size

Reply:
need to revert it back. We can't send more than seg size

Reply:
right, reverted